launch/model_endpoint.py

```diff
@@ -216,8 +218,30 @@ def get(self) -> EndpointResponse:
 class Endpoint:
     """An abstract class that represent any kind of endpoints in Scale Launch"""
 
-    def __init__(self, model_endpoint: ModelEndpoint):
+    def __init__(self, model_endpoint: ModelEndpoint, client):
         self.model_endpoint = model_endpoint
+        self.client = client
+
+    def _update_model_endpoint(self):
+        resp = self.client.connection.get(
+            os.path.join(ENDPOINT_PATH, self.model_endpoint.name)
+        )
+        self.model_endpoint = ModelEndpoint.from_dict(resp)
+
+    def status(self):
+        """Gets the status of the Endpoint."""
+        self._update_model_endpoint()
+        return self.model_endpoint.status
+
+    def resource_settings(self):
```

Comment:
Let's add return type annotations.

Reply:
Also, related: we'll want to regenerate the documentation and upload it to both the external and internal doc sites.

Reply:
currently this `endpoint` class is not covered in the external doc, I think it's because this class was just an abstract with no actual methods. I guess we'd like to add it to the doc if people are ok with this? 

btw in the internal doc, there's only `InternalLaunchClient`, no other endpoint related classes are there, do we want to add all of them?

Reply:
> currently this `endpoint` class is not covered in the external doc, I think it's because this class was just an abstract with no actual methods. I guess we'd like to add it to the doc if people are ok with this?

Yeah I think that's fine. I don't think we'll necessarily want to stick with these exact types going forward, but regardless, we should document whatever we have, since the user is receiving objects of this type regardless.

> btw in the internal doc, there's only InternalLaunchClient, no other endpoint related classes are there, do we want to add all of them?

The internal client should be subclassing this one, so it should inherit the methods, docstrings, etc. Let's verify this though.